### PR TITLE
enclave-tls: fix build warnings

### PR DESCRIPTION
**Pull request by haosanzi**
Fixes: #752
Signed-off-by: Shirong Hao <shirong@linux.alibaba.com>

### CODE REVIEW

File: enclave-tls/src/crypto_wrappers/wolfcrypt/gen_cert.c [hunk 1]
```diff
@@ -35,39 +35,41 @@ wolfcrypt_gen_cert(crypto_wrapper_ctx_t *ctx,
...
 		assert(sizeof(crt.iasSig) >= epid->ias_report_signature_len);
 		memcpy(crt.iasSig, epid->ias_report_signature, epid->ias_report_signature_len);
-		crt.iasSigSz = epid->ias_report_signature_len;
+		crt.iasSigSz = (int)epid->ias_report_signature_len;
 	} else if (!strcmp(cert_info->evidence.type, "sgx_ecdsa")) {
 		ecdsa_attestation_evidence_t *ecdsa = &cert_info->evidence.ecdsa;
 
 		memcpy(crt.quote, ecdsa->quote, ecdsa->quote_len);
-		crt.quoteSz = ecdsa->quote_len;
+		crt.quoteSz = (int)ecdsa->quote_len;
```
Comment (jiazhang0) [inline enclave-tls/src/crypto_wrappers/wolfcrypt/gen_cert.c:55]: quoteSz and other fields could be changed to unsigned type?

File: enclave-tls/src/include/enclave-tls/log.h [hunk 2]
```diff
@@ -10,6 +10,9 @@
 #ifdef SGX_ENCLAVE
   #include <enclave-tls/sgx.h>
 #endif
+#ifdef SGX_PRINT
```
Comment (jiazhang0) [inline enclave-tls/src/include/enclave-tls/log.h:13]: Why not only using SGX_ENCLAVE to resolve this problem?

File: enclave-tls/src/tls_wrappers/wolfssl-sgx/ecalls.c [hunk 3]
```diff
@@ -57,13 +57,13 @@ tls_wrapper_err_t ecall_wolfssl_cleanup(tls_wrapper_ctx_t *ctx)
 size_t recv(int sockfd, void *buf, size_t len, int flags)
 {
 	size_t ret;
-	int sgxStatus = ocall_recv(&ret, sockfd, buf, len, flags);
+	ocall_recv(&ret, sockfd, buf, len, flags);
```
Comment (jiazhang0) [inline enclave-tls/src/tls_wrappers/wolfssl-sgx/ecalls.c:60]: The check to sgxStatus is still necessary.

**Review by jiazhang0 — CHANGES_REQUESTED** [review]

File: enclave-tls/rules/sgx_env.mk [hunk 4]
```diff
@@ -112,7 +112,7 @@ else
   sgx_common_flags += -O2
 endif
 sgx_common_flags += \
-  -DSGX_ENCLAVE -I$(SGX_SDK)/include \
+  -I$(SGX_SDK)/include \
```
Comment (jiazhang0) [inline enclave-tls/rules/sgx_env.mk:115]: SGX_ENCLAVE should be a common flag used to identify current build context, and here is a good place to define it.

File: enclave-tls/src/sgx/Makefile [hunk 5]
```diff
@@ -23,7 +23,7 @@ Enclave_C_Files := \
   ecalls.c ../enclave_quotes/sgx-ecdsa/ecalls.c \
   ../tls_wrappers/wolfssl-sgx/ecalls.c \
   ../crypto_wrappers/wolfcrypt-sgx/ecalls.c
-Enclave_Cflags += -DSGX_ENCLAVE
+Enclave_Cflags += -DSGX_ENCLAVE -DWOLFSSL_SGX_WRAPPER
```
Comment (jiazhang0) [inline enclave-tls/src/sgx/Makefile:26]: wolfssl has defined "WOLFSSL_SGX" for trusted codes and it is defined in . Is it not enough?

I see you are trying to use WOLFSSL_SGX_WRAPPER as an identify of trusted enclave codes.

Comment (haosanzi) [reply enclave-tls/src/sgx/Makefile]: `WOLFSSL_SGX`  can use to replace `SGX_ENCLAVE` which is for trusted codes.
`WOLFSSL_SGX_WRAPPER` is used to distinguish different code implementations of `wolfssl` and `wolfssl-sgx`(untrusted code).
For example, the code in `un_negotiate.c` is included in `wolfssl` and `wolfssl-sgx` (untrusted code, ocalls.c) with different implemetation.

**Review by jiazhang0 — APPROVED** [review]

**Pull request merged**